### PR TITLE
Updated Plot Api

### PR DESCRIPTION
**Pull request by abhishekpratapa**
- Made Plot api plumbing for the show method
- Updated .show() method on Jupyter Notebook
- Updated Tooltips on Jupyter Notebook
- Created doc string for public methods of Plot Object

### CODE REVIEW

File: src/unity/python/turicreate/visualization/plot.py [hunk 1]
```diff
@@ -7,22 +7,99 @@
 import json as _json
 
 class Plot(object):
+    """
+    An immutable, object representative of a visualization.
```
Comment (znation) [inline src/unity/python/turicreate/visualization/plot.py]: s/representative/representation

Comment (znation) [reply src/unity/python/turicreate/visualization/plot.py]: And you don't need a comma between immutable and object.

File: src/unity/python/turicreate/visualization/plot.py [hunk 2]
```diff
@@ -7,22 +7,99 @@
...
 
+    Notes
+    -----
+    - A plot object is returned through a proxy via the SFrame's or SArray's .plot() method
```
Comment (znation) [inline src/unity/python/turicreate/visualization/plot.py]: The `through a proxy` part doesn't seem relevant to users. You can just say a plot object is returned from the `plot` method on Turi Create data structures.

File: src/unity/python/turicreate/visualization/plot.py [hunk 3]
```diff
@@ -7,22 +7,99 @@
...
-        import os
-        if sys.platform != 'darwin' and sys.platform != 'linux2':
-             raise NotImplementedError('Visualization is currently supported only on macOS and Linux.')
+        """
+        A method for displaying the Plot object
+
+        Notes
+        -----
+        - In a Jupyter Notebook the .show() method displays an inline plot
+        - In any other environment, the .show() method displays a native GUI
```
Comment (znation) [inline src/unity/python/turicreate/visualization/plot.py]: How about `launches a native GUI and displays the plot there`?

File: src/unity/python/turicreate/visualization/plot.py [hunk 4]
```diff
@@ -7,22 +7,99 @@
...
-    def save_vega(self, filepath, include_data=True):
+        """
+        display = False
+        try:
+            if(get_ipython().__class__.__name__ == "ZMQInteractiveShell"):
+                self._repr_javascript_()
+                display = True
+            else:
+                pass
+        except:
```
Comment (znation) [inline src/unity/python/turicreate/visualization/plot.py]: It would be better to catch the specific exception that happens when we're not in a Jupyter Notebook (and allow other exceptions to get raised from here).

File: src/unity/python/turicreate/visualization/plot.py [hunk 5]
```diff
@@ -7,22 +7,99 @@
...
+        >>> plt.show()
 
-    def save_vega(self, filepath, include_data=True):
+        """
+        display = False
+        try:
+            if(get_ipython().__class__.__name__ == "ZMQInteractiveShell"):
+                self._repr_javascript_()
+                display = True
+            else:
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/visualization/plot.py]: This else-block seems to have no purpose. So you probably want to delete it.

File: src/unity/python/turicreate/visualization/plot.py [hunk 6]
```diff
@@ -7,22 +7,99 @@
...
+                pass
+        except NameError:
+            pass
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/visualization/plot.py]: Why is it ok to ignore this error?

File: src/unity/python/turicreate/visualization/plot.py [hunk 7]
```diff
@@ -7,22 +7,99 @@
...
+        finally:
+            if not display:
+                import sys
+                import os
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/visualization/plot.py]: Doesn't look like you're using this import.

File: src/unity/python/turicreate/visualization/plot.py (hunk 6 above)
Comment (abhishekpratapa) [reply src/unity/python/turicreate/visualization/plot.py]: Using exceptions for control flow

Comment (TobyRoseman) [reply src/unity/python/turicreate/visualization/plot.py]: Sounds reasonable (I think). Please add a comment.
